cylc/flow/platforms.py

```diff
@@ -457,3 +457,20 @@ def is_platform_with_target_in_list(install_target, distinct_platforms_list):
     """Determines whether install target is in the list of platforms"""
     for distinct_platform in distinct_platforms_list:
         return install_target == distinct_platform['install target']
+
+
+def get_all_platforms_for_install_target(install_target):
+    """Return list of platform dictionaries for given install target."""
+    platforms = []
+    all_platforms = glbl_cfg(cached=True).get(['platforms'], sparse=False)
+    for k, v in all_platforms.iteritems():
+        if (v.get('install target', k) == install_target):
+            v['name'] = k
```

Comment:
Platform name is not already set, since the platform name comes from a different tier of the config.

Reply:
This isn't the nicest thing to do as it modifies the global config itself.

Reply:
That sort of thing seems to happen a lot in `platforms.py`. E.g. the reason I though `name` was already set is because `platform_from_name()` does that: https://github.com/cylc/cylc-flow/blob/951f265636ee9f828248f832c30ed523ef164b35/cylc/flow/platforms.py#L201-L202

Reply:
That example doesn't mutate the global config itself because a few lines earlier it calls `deepcopy` to duplicate the data, it is the copy that it mutates:

https://github.com/cylc/cylc-flow/blob/951f265636ee9f828248f832c30ed523ef164b35/cylc/flow/platforms.py#L189

Reply:
Trouble is, `platform_from_name()`  makes calls out to `deepcopy`, as such, `platform['name']` is not set by the time we need it here. I'll put up a code change to prevent my change modifying the global config but do we need the deep copies here? I think we could revisit this function, removing the `deepcopy` calls, which would have the following benefits: 

- more efficient given there are a fair few calls to it
- we would be able to set the platform name, and things like install target, early, for use elsewhere  

One of the comments with a `deepcopy` states that it is used because:

>   Deepcopy prevents contaminating platforms with data
>   from other platforms matching platform_name_re  

Unless I'm missing something, I don't think this is an issue in a for loop - we can set values for keys in each element of a for loop independently?

Reply:
I think Oliver mentioned yesterday in our meeting that we could instead try setting things like `name` on the initial load of the config only